ReactAndroid/src/main/java/com/facebook/react/bridge/BaseJavaModule.java

```diff
@@ -49,17 +54,35 @@
  */
 public abstract class BaseJavaModule implements NativeModule {
   // taken from Libraries/Utilities/MessageQueue.js
-  static final public String METHOD_TYPE_ASYNC = "async";
-  static final public String METHOD_TYPE_PROMISE= "promise";
+  private static final String METHOD_TYPE_ASYNC = "async";
+  private static final String METHOD_TYPE_PROMISE= "promise";
   static final public String METHOD_TYPE_SYNC = "sync";
+  private final List<? extends ArgumentExtractor.Factory> mArgumentExtractorFactories;
+
+  public BaseJavaModule() {
+    this(Collections.<ArgumentExtractor.Factory>emptyList());
+  }
 
-  private static abstract class ArgumentExtractor<T> {
+  public BaseJavaModule(List<? extends ArgumentExtractor.Factory> argumentExtractorFactories) {
```

Comment:
Let's go ahead and change to using the method instead. I realize I feel strongly about this for a few reasons; even if I agree on that inheritance is not always a good solution, however in this case it's really implementing an interface (even if we provide a simplifying base class). The method also gives us more lazy behavior which is something were trying hard to move to, it also matches how this class already works.